tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1401,3 +1401,96 @@ def test_timeout_and_cac_card_ejection():
 
     :expectedresults: Satellite should terminate the session after mentioned timeout in setting
     """
+
+
+@tier2
+def test_positive_end_to_end_open_ldap_authsource(
+    test_name, session, ldap_tear_down, open_ldap_data
+):
+    """Create OpenLDAP auth_source with org and loc assigned.
+
+    :id: f2fa0678-ff2f-11ea-b081-d46d6dd3b5b2
+
+    :steps:
+        1. Create a new LDAP Auth source with OpenLDAP, provide organization and
+           location information.
+        2. Fill in all the fields appropriately for OpenLDAP.
+
+    :expectedresults: Whether creating LDAP Auth source with OpenLDAP and
+        associating org and loc is successful.
+    """
+    org = entities.Organization().create()
+    loc = entities.Location().create()
+    ldap_auth_name = gen_string('alphanumeric')
+    with session:
+        session.ldapauthentication.create(
+            {
+                'ldap_server.name': ldap_auth_name,
+                'ldap_server.host': open_ldap_data['ldap_hostname'],
+                'ldap_server.ldaps': False,
+                'ldap_server.server_type': LDAP_SERVER_TYPE['UI']['posix'],
+                'account.account_name': open_ldap_data['ldap_user_name'],
+                'account.password': open_ldap_data['ldap_user_passwd'],
+                'account.base_dn': open_ldap_data['base_dn'],
+                'account.onthefly_register': True,
+                'locations.resources.assigned': [loc.name],
+                'organizations.resources.assigned': [org.name],
+            }
+        )
+        session.organization.select(org_name=org.name)
+        session.location.select(loc_name=loc.name)
+        assert session.ldapauthentication.read_table_row(ldap_auth_name)['Name'] == ldap_auth_name
+        ldap_source = session.ldapauthentication.read(ldap_auth_name)
+        assert ldap_source['ldap_server']['name'] == ldap_auth_name
+        assert ldap_source['ldap_server']['host'] == open_ldap_data['ldap_hostname']
+        assert ldap_source['ldap_server']['port'] == '389'
+    user_name = open_ldap_data['ldap_user_name'].split(',')[0][3:]
```

Comment:
`.split(',')[0][3:]` - could we make that less cryptic? Why user name must be split on `,`? Why are we dropping first three characters?

Reply:
I'm, using the username from this property `username=cn=<username>,ou=users,dc=<domain>,dc=com` rather saving same username in property